packages/lodestar/src/db/api/beacon/stateContextCache.ts

```diff
@@ -1,20 +1,24 @@
 import {ByteVector, toHexString, TreeBacked} from "@chainsafe/ssz";
 import {BeaconState} from "@chainsafe/lodestar-types";
 import {EpochContext} from "@chainsafe/lodestar-beacon-state-transition";
+import {StateRepository} from "./repositories/state";
 
 export interface ITreeStateContext {
   state: TreeBacked<BeaconState>;
   epochCtx: EpochContext;
 }
 
 /**
- * In memory cache of BeaconState and connected EpochContext
+ * In memory cache of BeaconState and connected EpochContext.
+ * It also forward calls to state repository.
  *
  * Similar API to Repository
  */
 export class StateContextCache {
```

Comment:
In-memory cache and StateRepository go in parallel. On checkpoint, in-memory cache is pruned but StateRepository is not.